File: api/Models/PhoneExpense.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Cretovale_api.Models
{
  public class PhoneExpense : IEntity
  {
    public Guid Id { get; set; }

    public Phone Phone { get; set; }

    public Expense Expense { get; set; }

    [Required]
    public int Year { get; set; }

    [Required]
    public int Month { get; set; }

    [Required]
    public decimal Value { get; set; }

    public DateTime DateCreated { get; set; }
    public DateTime DateUpdated { get; set; }
  }
}

File: client/src/app/shared/models/phone-expense.ts
import { Phone } from './phone';
import { Expense } from './expense';

export class PhoneExpense {
    constructor(public id?: string,
                public phone?: Phone,
                public expense?: Expense,
                public year?: number,
                public month?: number,
                public value?: number,
                public name?: string,
                public category?: string) {
    }
}
